internal/backend/remote-state/http/client.go

```diff
@@ -42,14 +42,13 @@ type httpClient struct {
 }
 
 func (c *httpClient) httpRequest(method string, url *url.URL, data *[]byte, what string) (*http.Response, error) {
-	// If we have data we need a reader
-	var reader io.Reader = nil
+	var body interface{}
 	if data != nil {
-		reader = bytes.NewReader(*data)
+		body = *data
```

Comment:
The code works differently in case of no body vs empty body. They are not the same thing at the moment. On the other hand, it always sets `Content-Type` to `application/json` and empty body is not a valid JSON, so this is unlikely a breaking change. Made the change.